### PR TITLE
HE - New game - Dice.

### PR DESCRIPTION
**Pull request by Dartellum**
Just need `dice_bet_amount: 10` (or whatever you want) in your yaml or call with `;crowns 20` to override yaml setting or not put one in.

### CODE REVIEW

File: crowns.lic [hunk 1]
```diff
@@ -0,0 +1,69 @@
+
```
Comment (rpherbig) [inline crowns.lic:1]: 💡 Needs documentation header

Comment (rpherbig) [inline crowns.lic:1]: ❌ Needs `custom_require`

Comment (Dartellum) [reply crowns.lic:1]: I will add tonight. This was “quick” for some to test. Formayor is one tester.

Comment (Dartellum) [reply crowns.lic:1]: Okay, will add tonight.

Comment (Dartellum) [reply crowns.lic:1]: Added.

Comment (Dartellum) [reply crowns.lic:1]: Added. Already added to Lich Repo scripting reference too.

**Comment by Dartellum** [comment]
Should I add to this one to grab money?

File: crowns.lic [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  [
+    { name: 'bet', regex: /\d+/, optional: true, description: 'Optional override for betting amount.' }
+  ]
+]
+
+custom_require.call(%w[events common-travel common])
+
+settings = get_settings
+junk_list = settings.darkbox_junk.map { |x| /#{x}/i }
+dice_bet_amount = settings.dice_bet_amount || args.bet
```
Comment (rpherbig) [inline crowns.lic:15]: `dice_bet_amount` needs to be in `base.yaml`

**Comment by rpherbig** [comment]
> Should I add to this one to grab money?

That's your call.

Since this is a new script, I can merge it whenever you're ready.

File: crowns.lic (hunk 2 above)
Comment (Dartellum) [reply crowns.lic:15]: Added to base.

**Comment by Dartellum** [comment]
Rewrote the script.

File: crowns.lic [hunk 3]
```diff
@@ -2,84 +2,128 @@
...
-    when /You don't have that many kronars!/
-      DRC.message("******* Out of money! ********")
+  def check_money
+    get_money if check_wealth('kronars') < 200 && @withdraw
+  end
+
+  def get_money
+    DRC.message("*** Heading to bank to money! ***")
+    DRCT.walk_to(16_315)
+    case DRC.bput("withdraw #{@dice_money_on_hand} platinum", 'we are not lending money at this time', 'The clerk counts out')
```
Comment (rpherbig) [inline crowns.lic:51]: ❓ Doesn't this result in something like `withdraw 10 platinum platinum`?

File: crowns.lic [hunk 4]
```diff
@@ -2,84 +2,128 @@
...
   end
 
-  if count = 6
-    pause 1 until Flags['no_more_bets']
-    Flags.reset('no_more_bets')
+  def play_dice
+    loop do
+      count = 0
+      @dice.each do |id|
+        case DRC.bput("bet #{@dice_bet_amount} on #{id}", /You bet/, /You can't bet right now./, /You don't have that many kronars!/)
```
Comment (rpherbig) [inline crowns.lic:83]: ❓ Is it sufficient to `bet 10 on {thing}` and not need to specify the denomination (i.e. `bet 10 platinum on {thing}`?

File: crowns.lic [hunk 5]
```diff
@@ -1,5 +1,5 @@
 =begin
-  Documentation: https://elanthipedia.play.net/Lich_script_repository#dice-game
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#Hollow_Eve_Dice_Game
```
Comment (rpherbig) [inline crowns.lic:2]: 💡 this is usually the name of the script, i.e. `crowns`

File: crowns.lic (hunk 4 above)
Comment (Dartellum) [reply crowns.lic:83]: You are betting coppers.

File: crowns.lic (hunk 5 above)
Comment (Dartellum) [reply crowns.lic:2]: I will change it around. I put the script name in the header of it. Wanted to make sure everyone knew was HE dice and not some dice game elsewhere.

File: crowns.lic (hunk 3 above)
Comment (Dartellum) [reply crowns.lic:51]: Yes, every time you dip below 200 coppers.

Comment (rpherbig) [reply crowns.lic:51]: Does that command work with the denomination in there twice? I'm just surprised, is all

Comment (Dartellum) [reply crowns.lic:51]: Let me test that particular piece. It might not. I copied over from another script and modified it.

**Comment by Dartellum** [comment]
Running a few more tests.

File: crowns.lic (hunk 3 above)
Comment (Dartellum) [reply crowns.lic]: Removed platinum from the line as it is in base. I like it in base because the person knows what it is grabbing.

**Comment by rpherbig** [comment]
✔️  Looks good, is this ready to be merged?

**Comment by Dartellum** [comment]
Yes, ready to merge. I have another update for all the games to use a consolidated `hollow_eve_junk:` list. This will come in a separate PR.

**Pull request merged**